### ISSUE TITLE
can't get the desired training result

### ISSUE THREAD

**Issue by yuqi1991** [TRI-ML/packnet-sfm#16]
Hi there,
   Thanks so much for sharing the training code, I tried to run with self-sup model over KITTI dataset, but it seems to have a weird result after running for a few epochs with pretrained model.

![Screenshot from 2020-05-19 13-46-18](https://user-images.githubusercontent.com/11883429/82290248-093a7380-99d9-11ea-8456-4ea2764134dc.png)

The loss get smaller during training, but the evaluation metrics of error get higher every epoch. Did you have ever encounter this situation ?

@AdrienGaidon-TRI @VitorGuizilini-TRI @spillai

**Comment by VitorGuizilini-TRI** [comment]
Which pretrained model are you using, and which config file?

**Comment by yuqi1991** [comment]
@VitorGuizilini-TRI 
Pretrained model: PackNet, Self-Supervised, 192x640, KITTI (K)
Config file: train_kitti.yaml

**Comment by VitorGuizilini-TRI** [comment]
Are you doing single GPU training, and with what batch size? The learning rates we specify are for 8 GPUs (as mentioned in the paper) and a batch size of 4, you might need to decrease them if you are doing single GPU training.

**Comment by yuqi1991** [comment]
@VitorGuizilini-TRI 
I use 8 GPU as well and have been running for more than 10 hours, the avg loss is stagnating at around 0.09.

**Comment by VitorGuizilini-TRI** [comment]
We regularly use that model for fine-tuning on KITTI and other datasets, and never observed that behavior. I will look deeper, but one last question: are you loading the checkpoint in the config file (checkpoint_path) or are you resuming from the checkpoint itself?

**Comment by yuqi1991** [comment]
I tried both training from scratch and loading the model by manually extracting the state dicts from the checkpoint.

**Comment by yuqi1991** [comment]
Another question about the semi-sup model, why did you guys disable the photometric loss ? I think it would be better to joint the photometric loss and the pointcloud loss together, since the point cloud is sparse on the re-projection plane.

**Comment by VitorGuizilini-TRI** [comment]
About fine-tuning on pretrained models, I don't know what to say, I just ran an experiment here with the same code on this repository and it looks alright. The loss starts already stable at ~0.073, and the metrics are also stable at the numbers we report. Try decreasing the learning rate to 0.00005, that's the value at the end of the training session (0.0002/4). Are you running inside our docker? 

![media_images_val-KITTI_raw-eigen_test_files-velodyne-544-inv_depth_27_d6fa87c8](https://user-images.githubusercontent.com/58576956/82567746-a9f07680-9b32-11ea-9142-504968995dfc.png)

About the semi-sup model, we are using the self-supervised photometric loss in addition to the supervised loss. You can set the ratio in the config file, if you set to 1.0 then it's purely supervised.

**Comment by yuqi1991** [comment]
yes, I was running inside the docker, and have just replaced the wandb logger with tensorboard for visualization. I will continue try look into it. So 0.073 is the best loss you get on your side right ? Thanks for your help.

**Comment by MingYang-buaa** [comment]
@yuqi1991 Hi, i'm newer to pytorch,  can you shared the code for visualization with tensorboard?

**Comment by yuqi1991** [comment]
@MingYang-buaa hi, since I already left the project, I can not provide you the source code, but maybe you can refer to the usage document here of  [tensorboard X](https://tensorboardx.readthedocs.io/en/latest/tutorial.html), it's quite easy to use.
